tests/tasks/lib/test_mnli.py

```diff
@@ -1,34 +1,321 @@
+            "the",
+            "hundred",
+            "dollar",
+            "range",
+        ],
+        "hypothesis": ["The", "tennis", "shoes", "have", "a", "range", "of", "prices."],
+        "label_id": 2,
+    },
+]
+
+FEATURIZED_TRAIN_EXAMPLE_0 = {
+    "guid": "train-0",
+    "input_ids": np.array([1, 4, 5, 6, 7, 8, 9, 10, 11, 2, 15, 13, 16, 17, 18, 19, 5, 6, 20, 2]),
+    "input_mask": np.array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]),
+    "segment_ids": np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]),
+    "label_id": 2,
+    "tokens": [
+        "<cls>",
+        "Conceptually",
+        "cream",
+        "skimming",
+        "has",
+        "two",
+        "basic",
+        "dimensions",
+        "-",
+        "<sep>",
+        "Product",
+        "and",
+        "geography",
+        "are",
+        "what",
+        "make",
+        "cream",
+        "skimming",
+        "work.",
+        "<sep>",
+    ],
+}
+
+
+def test_featurization_of_task_data():
+    # Test reading the task-specific toy dataset into examples.
+    task = create_task_from_config_path(
+        os.path.join(os.path.dirname(__file__), "resources/mnli.json"), verbose=False
     )
-    assert np.array_equal(
-        featurized_example.input_mask,
-        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0],
+    # Test getting train, val, and test examples. Only the contents of train are checked.
+    train_examples = task.get_train_examples()
+    val_examples = task.get_val_examples()
+    test_examples = task.get_test_examples()
+    for train_example_dataclass, raw_example_dict in zip(train_examples, TRAIN_EXAMPLES):
+        assert train_example_dataclass.to_dict() == raw_example_dict
+    assert val_examples
+    assert test_examples
+
+    # Testing conversion of examples into tokenized examples
+    # the dummy tokenizer requires a vocab — using a Counter here to find that vocab from the data:
+    token_counter = Counter()
+    for example in train_examples:
+        token_counter.update(example.premise.split())
+        token_counter.update(example.hypothesis.split())
+    token_vocab = list(token_counter.keys())
+    tokenizer = SimpleSpaceTokenizer(vocabulary=token_vocab)
+    tokenized_examples = [example.tokenize(tokenizer) for example in train_examples]
+    for tokenized_example, expected_tokenized_example in zip(
+        tokenized_examples, TOKENIZED_TRAIN_EXAMPLES
+    ):
+        assert tokenized_example.to_dict() == expected_tokenized_example
+
+    # Testing conversion of a tokenized example to a featurized example
+    train_example_0_length = len(tokenized_examples[0].premise) + len(
+        tokenized_examples[0].hypothesis
     )
-    assert np.array_equal(
-        featurized_example.segment_ids,
-        [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0],
+    feat_spec = model_resolution.build_featurization_spec(
+        model_type="bert-", max_seq_length=train_example_0_length
```

Comment:
minor point -> needing`model_type="bert-"` highlights that we may need to create an enum or using an existing enum for model_type instead of a string

Reply:
+1

In this instance you could dodge providing the string by getting a `feat_spec` from the mock tokenizer [here](https://github.com/jiant-dev/jiant/blob/81a8a3d4be6a01802005347c5ef3ec93a9a78c7a/jiant/utils/testing/tokenizer.py#L31).

Reply:
Will include that as part of a new review (will change for sst too).